open_event/static/js/jquery/jquery.codezero.js

```diff
@@ -0,0 +1,221 @@
+    var h2 = $div2.outerHeight(true);
+    var w2 = $div2.outerWidth(true);
+    var b2 = y2 + h2;
+    var r2 = x2 + w2;
+    return !(b1 < y2 || y1 > b2 || r1 < x2 || x1 > r2);
+}
+
+
+/**
+ * Log an error message to the console
+ * @param {string} message The message
+ * @param {Object} [ref] a reference object that will be printed along with the message
+ */
+function logError(message, ref) {
+    if (_.isUndefined(ref)) {
+        console.log("[E] " + message);
+    } else {
+        console.log("[E] " + message + ". Ref ↴");
+        console.log(ref);
+    }
+}
+
+/**
+ * Log a debug message to the console
+ * @param {string} message The message
+ * @param {Object} [ref] a reference object that will be printed along with the message
+ */
+function logDebug(message, ref) {
+    if (_.isUndefined(ref)) {
+        console.log("[D] " + message);
+    } else {
+        console.log("[D] " + message + ". Ref ↴");
+        console.log(ref);
+    }
+}
+
+/**
+ * Detect if a variable is undefined or null
+ * @param {*} variable The variable
+ * @returns {boolean} The result of the check
+ */
+function isUndefinedOrNull(variable) {
+    return (_.isUndefined(variable) || _.isNull(variable));
+}
+
+/**
+ * Create a material design snackbar to display simple notifications
+ */
+var createSnackbar = (function () {
+    var previous = null;
+
+    return function (message, actionText, action, delay) {
+        if (previous) {
+            previous.dismiss();
+        }
+
+        if (typeof actionText === 'undefined' || actionText == null) {
+            actionText = "Dismiss";
+        }
+
+        if (typeof delay === 'undefined' || delay == null) {
+            delay = 5000;
+        }
+        var snackbar = document.createElement('div');
+        snackbar.className = 'paper-snackbar';
+        snackbar.dismiss = function () {
+            this.style.opacity = 0;
+        };
+        var text = document.createTextNode(message);
+        snackbar.appendChild(text);
+        if (actionText) {
+            var hasAction = true;
+            if (!action) {
+                action = snackbar.dismiss.bind(snackbar);
+                hasAction = false;
+            }
+            var actionButton = document.createElement('button');
+            actionButton.className = 'action';
+            actionButton.innerHTML = actionText;
+            if (hasAction) {
+                actionButton.addEventListener('click', function () {
+                    action();
+                    snackbar.dismiss.bind(snackbar);
+                });
+            } else {
+                actionButton.addEventListener('click', action);
+            }
+
+            snackbar.appendChild(actionButton);
+        }
+        setTimeout(function () {
+            if (previous === this) {
+                previous.dismiss();
+            }
+        }.bind(snackbar), delay);
+
+        snackbar.addEventListener('transitionend', function (event, elapsed) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['elapsed' is defined but never used.](https://www.codacy.com/app/triplez/open-event-orga-server/file/2674855960/issues/source?bid=3233864&fileBranchId=3324174#l205)